contrib/epee/src/mlocker.cpp

```diff
@@ -160,18 +181,30 @@ namespace epee
 
   size_t mlocker::get_num_locked_pages()
   {
+#if defined __EMSCRIPTEN__
+    return 0;
+#endif
```

Comment:
Why add all of these when you can just not define HAVE_MLOCK ? Is this somehow not working ?

Reply:
It is somehow not working (i.e. something is calling one of the mlocker class functions).  I will look into it further.

Reply:
`mlocker` functions are being called by using the `mlocked` struct like [here](https://github.com/monero-project/monero/blob/39e18902763a5fd6f355294d8b135c8f5bc2ec99/src/wallet/wallet2.cpp#L3923) in `wallet2::setup_keys()`, so currently, individual functions in `mlocker` need guarded.

Reply:
I don't understand the implication. Can you explain ?

Reply:
Execution in WebAssembly will fail on `CRITICAL_REGION_LOCAL(mutex())` in mlocker.cpp even if HAVE_MLOCK is not defined because the `mlocked` struct is used externally without checking for this definition.  The current checks in question prevent that.  Do you prefer another way?